test/core.jl

```diff
@@ -784,7 +784,7 @@ end
 mutable struct Type11167{T,N} end
 function count11167()
     let cache = Type11167.body.body.name.cache
-        return sum(i -> isassigned(cache, i), 0:length(cache))
+        return sum(i -> cache[i] !== nothing, 1:length(cache), init=0)
```

Comment:
```suggestion
        return count(e->e !== nothing, cache)
```
?

Reply:
Or maybe `count(!=(nothing), cache)` :)

Reply:
nothing usually should not be compared with `==`, but it turns out we may have a shorter predicate for this (`!isnothing`)